docs/source/conf.py

```diff
@@ -53,9 +53,23 @@
 extensions = [
     'sphinx.ext.autodoc',
     'sphinx.ext.napoleon',
-    'sphinx.ext.intersphinx'
+    'sphinx.ext.intersphinx',
+    'sphinx_gallery.gen_gallery'
 ]
 
+from sphinx_gallery.sorting import ExplicitOrder
+
+sphinx_gallery_conf = {
```

Comment:
Sphinx-gallery can execute the tutorial code for you while building the docs. I think that is a fantastic option and it would catch syntax and import errors in the docs for free.

There are 2 ways to do it:

1) We can add a regex to actually select and execute the tutorial code. 
2) We can change the file names to start with the prefix "plot_" and it will automatically execute the tutorials and catch syntax and import errors. For example "docs/gallery/general/file.py" --> "docs/gallery/general/plot_file.py". I think that is sphinx-gallery's convention.

See https://sphinx-gallery.readthedocs.io/en/latest/advanced_configuration.html#parsing-and-executing-examples-based-on-matching-patterns

In the future it would be great to assert some cases against tutorial code. I would be interested in doing that work but that is another PR.

Reply:
The examples are already executed in the test script here:
https://github.com/NeurodataWithoutBorders/pynwb/pull/411/files#diff-b284a28710cce90d9d9be3a7f4cabc8eR65

Reply:
I added a dummy import https://github.com/NeurodataWithoutBorders/pynwb/blob/e67b8346cf9c7a548c54e1c9f7a195ef810cafa1/docs/gallery/domain/ophys.py#L21 here such as 
```python
import foobar
```
and it did not raise en error till I change the filename to plot_ophys.py.

Can you verify?

Reply:
Oh, ok I see now. It will be executed during the tests. The thing that I am talking about executes them when you 
```
make html
```
Ok, I am fine as long as we execute at some point.

Reply:
I understand now. We're talking about two different modes of execution. I've updated test.py so that it executes when we run the test suite (on CircleCI). You're referring to sphinx-gallery executing the scripts when building the docs (which would take place on RTD).

Do we want them in both places? I'm fine with both or just Circle

Reply:
FYI: https://sphinx-gallery.readthedocs.io/en/latest/advanced_configuration.html#build-pattern

Reply:
Yes, exactly. I am fine with the way it is now. Thank you!!